### PR TITLE
pysmi 1.5.9

### PR DESCRIPTION
**Pull request by SC426**
**Destination channel:** defaults

### Links

- [PKG-6504](https://anaconda.atlassian.net/browse/PKG-6504)
- [Upstream repository](https://github.com/etingof/pysmi)
- Relevant dependency PRs:
  - `pysmi` 👉  `pysnmp'

### Explanation of changes:

- update version
- remove noarch
- update build script and dependencies
- add pip check
- other linter fixes

### Notes
- not sure what to do about the broken snmplabs.com link

[PKG-6504]: https://anaconda.atlassian.net/browse/PKG-6504?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -35,17 +37,21 @@ test:
...
     - pysmi.writer
+  requires:
+    - pip
+  commands:
+    - pip check
 
 about:
-  home: http:/pysmi.sf.net
-  license: BSD 2-Clause
+  home: https://pysmi.sourceforge.net/
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
  home: https://www.pysnmp.com/pysmi/
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -35,17 +37,21 @@ test:
...
+  license: BSD-2-Clause
   license_family: BSD
   license_file: LICENSE.rst
   summary: SNMP SMI/MIB Parser
   description: 'A pure-Python implementation of SNMP/SMI MIB parsing and conversion library.'
-  doc_url: http://snmplabs.com/pysmi/#documentation
+  doc_url: https://snmplabs.com/pysmi/#documentation
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
  doc_url: https://docs.lextudio.com/pysmi/
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -35,17 +37,21 @@ test:
...
   dev_url: https://github.com/etingof/pysmi
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
  dev_url: https://github.com/lextudio/pysmi
```

**Review by cbouss — COMMENTED** [review]
The project owner lost their original domain. So the home doc and dev urls have changed.

**Review by ELundby45 — APPROVED** [review]

**Review by cbouss — APPROVED** [review]

**Pull request merged**